### ISSUE TITLE
Wrong variables in FootnoteExtension.py

### ISSUE THREAD

**Issue by didiertremblay** [obsidian-html/obsidian-html#553]
For my file conversion I got an error where `i` is undefined. Following the code is seems the line 167 contains a `i` instead of `rid`. Putting `rid` appear to fix my issue.

Original code
```python
    def RestoreCodeSections(self, block):
        """Undo the action of StripCodeSections."""
        for rid in self.codeblocks.keys():
            value = self.codeblocks[rid]
            block = block.replace(f'%%%codeblock-placeholder-{i}%%%', f"```{value}```\n")
        for rid in self.codelines.keys():
            value = self.codelines[rid]
            block = block.replace(f'%%%codeline-placeholder-{rid}%%%', f"`{value}`")  
        return block
```

Correction
```python
    def RestoreCodeSections(self, block):
        """Undo the action of StripCodeSections."""
        for rid in self.codeblocks.keys():
            value = self.codeblocks[rid]
            block = block.replace(f'%%%codeblock-placeholder-{rid}%%%', f"```{value}```\n")
        for rid in self.codelines.keys():
            value = self.codelines[rid]
            block = block.replace(f'%%%codeline-placeholder-{rid}%%%', f"`{value}`")  
        return block
```

**Comment by dwrolvink** [comment]
Thank you! The fix works.

Though then I noticed that code blocks don't natively work in python-markdown, unless it's in the root (so not indented like under a list or in a footnote).

I managed to work around this by converting codeblocks that use backticks to indented codeblocks. This does mean that the code highlighting won't work, but it's the only way that I can get codeblocks to work without the newlines being cut out.

Let me know if this change works for you.
